deps/rabbit/src/rabbit_classic_queue.erl

```diff
@@ -127,7 +127,7 @@ is_recoverable(Q) when ?is_amqqueue(Q) ->
     %% record if it is a mirrored queue (such info is now obtained from
     %% the policy). Thus, we must check if the local pid is alive
     %% - if the record is present - in order to restart.
-    (mnesia:read(rabbit_queue, amqqueue:get_name(Q), read) =:= []
+    (not rabbit_amqqueue:exists(amqqueue:get_name(Q))
```

Comment:
I don't have it clear either, maybe none but reasoning about transient mirrored records is hard. I'm restoring the original behaviour by introducing `rabbit_db_queue:consistent_exists/1` just for this. It can be removed with transient entities.

Reply:
In this part of the code, especially given the comment above, I'd keep the original (more consistent) read.

Reply:
Yes, that is done